packages/driver/src/cy/commands/asserting.ts

```diff
@@ -34,12 +34,25 @@ export default function (Commands, Cypress, cy, state) {
   const shouldFnWithCallback = function (subject, fn) {
     state('current')?.set('followedByShouldCallback', true)
 
+    const commandEnqueued = (obj) => {
+      $errUtils.throwErrByPath(
+        'should.command_inside_should', {
+          args: { action: obj.name },
+        },
+      )
+    }
+
     return Promise
     .try(() => {
       const remoteSubject = cy.getRemotejQueryInstance(subject)
 
+      Cypress.once('command:enqueued', commandEnqueued)
```

Comment:
Users would have no issues using `cy` commands inside of `should` as long as the command doesn't `verifyUpcomingAssertions` right? So lots of commands like `cy.intercept()`, `cy.setCookies`, etc. would all work in `.should`... It does seem like this has the potential to break a lot of tests that are innocently using these commands in `.should`. I'd propose merging this to `11.0.0` as a breaking change.

Reply:
Commands that don't `verifyUpcomingAssertions` still create chainers and can potentially modify the subject; The undefined behavior is more about the command queue (adding commands to it multiple times when .should() retries) and the command log (logging multiple times, each time the command is re-added to the queue) and the subject (when commands resolve as part of another command, the new subject replaces the old one - this causes issues from inside assertions).

"retries inside of retries" isn't the issue, it's "calling cypress commands repeatedly" that's the problem.

Reply:
I think I'm not totally getting it, but in your opinion, are there testing patterns users use that are sane-looking that would break after this change?

Reply:
In this particular case, I care more about munging the subject. This is desired behavior:

```
cy.then(() => {
  cy.wrap('foo')
}).should('equal', 'foo')
```

But this is weird behavior: 

```
cy.window().should(subject => {
  expect(subject).to.equal('foo')
  cy.wrap('foo')
})
```

They are both logical consequences of "chainers created inside of other commands become the new subject", but the second one is just *weird*. We should provide guard rails and guide our users towards tests that make sense.

Reply:
I don't think there are any sane-looking tests that will break. But also, users do all sorts of things we don't really expect?

I'm inclined to agree with marking this breaking and putting it out as part of 11.x - that was my initial intention, but `break:'` apparently isn't a semantic PR title.